airbyte-config/init/src/main/java/io/airbyte/config/init/LocalDefinitionsProvider.java

```diff
@@ -28,41 +21,48 @@
-  private final static String SPEC = "spec";
+  private static final String LOCAL_CONNECTOR_CATALOG_PATH = CatalogDefinitionsConfig.getLocalConnectorCatalogPath();
 
-  private Map<UUID, StandardSourceDefinition> sourceDefinitions;
-  private Map<UUID, StandardDestinationDefinition> destinationDefinitions;
+  public CombinedConnectorCatalog getLocalDefinitionCatalog() {
+    try {
+      final URL url = Resources.getResource(LOCAL_CONNECTOR_CATALOG_PATH);
+      final String jsonString = Resources.toString(url, StandardCharsets.UTF_8);
+      final CombinedConnectorCatalog catalog = Jsons.deserialize(jsonString, CombinedConnectorCatalog.class);
+      return catalog;
 
-  // TODO inject via dependency injection framework
-  private final Class<?> seedResourceClass;
-
-  public LocalDefinitionsProvider(final Class<?> seedResourceClass) throws IOException {
-    this.seedResourceClass = seedResourceClass;
+    } catch (final Exception e) {
+      throw new RuntimeException("Failed to fetch local catalog definitions", e);
+    }
+  }
 
-    // TODO remove this call once dependency injection framework manages object creation
-    initialize();
+  public Map<UUID, StandardSourceDefinition> getSourceDefinitionsMap() {
+    final CombinedConnectorCatalog catalog = getLocalDefinitionCatalog();
+    return catalog.getSources().stream().collect(Collectors.toMap(
+        StandardSourceDefinition::getSourceDefinitionId,
+        source -> source.withProtocolVersion(
+            AirbyteProtocolVersion.getWithDefault(source.getSpec() != null ? source.getSpec().getProtocolVersion() : null).serialize())));
   }
 
-  // TODO will be called automatically by the dependency injection framework on object creation
-  public void initialize() throws IOException {
-    this.sourceDefinitions =
-        parseDefinitions(this.seedResourceClass, SeedType.STANDARD_SOURCE_DEFINITION.getResourcePath(), SeedType.SOURCE_SPEC.getResourcePath(),
-            SeedType.STANDARD_SOURCE_DEFINITION.getIdName(), SeedType.SOURCE_SPEC.getIdName(), StandardSourceDefinition.class);
-    this.destinationDefinitions = parseDefinitions(this.seedResourceClass, SeedType.STANDARD_DESTINATION_DEFINITION.getResourcePath(),
-        SeedType.DESTINATION_SPEC.getResourcePath(), SeedType.STANDARD_DESTINATION_DEFINITION.getIdName(), SeedType.DESTINATION_SPEC.getIdName(),
-        StandardDestinationDefinition.class);
+  public Map<UUID, StandardDestinationDefinition> getDestinationDefinitionsMap() {
+    final CombinedConnectorCatalog catalog = getLocalDefinitionCatalog();
+    return catalog.getDestinations().stream().collect(
+        Collectors.toMap(
+            StandardDestinationDefinition::getDestinationDefinitionId,
+            destination -> destination.withProtocolVersion(
+                AirbyteProtocolVersion.getWithDefault(
+                    destination.getSpec() != null
+                        ? destination.getSpec().getProtocolVersion()
```

Comment:
The sextuple indent makes me think this might be worth breaking out into its own function. One stylistic nitpik you can feel free to ignore would be to use `Optional` again here
```
destination -> Optional.ofNullable(destination.getSpec()).map(spec::getProtocolVersion).orElse(null)
```